### ISSUE TITLE
Source maps would help debug some classes of problems

### ISSUE THREAD

**Issue by canadaduane** [gohyperr/hecs#13]
Occasionally I do something dumb (ok, quite often) and end up in some code path that HECS didn't anticipate, throwing an error that is hard to debug.

For example:

```

```
<img width="897" alt="image" src="https://user-images.githubusercontent.com/129/97503817-5ed95b80-193b-11eb-883f-d4be57271697.png">

<img width="1225" alt="image" src="https://user-images.githubusercontent.com/129/97503795-52550300-193b-11eb-9aef-00f84dfb2d9a.png">

Adding source maps to the `dist/` folder would help here (maybe).

**Comment by ashconnell** [comment]
Yeah I haven't gotten around to it yet. Personally I don't run into this much because i've been using it for a long time and must know how to avoid it, but i understand people just picking HECS up are likely to stumble into this quite often.

I might be able to look at adding source maps in the next week or two, it's probably not too hard. 

Do you have a snippet that reproduces an error like the one above?

**Comment by canadaduane** [comment]
> Do you have a snippet that reproduces an error like the one above?

Sure, here's my latest dumb mistake :)

Inside a `System`'s update method:

```
const transform = entity.get(Transform);
transform.position.add(undefined)
```

**Comment by ashconnell** [comment]
These are the steps I took to create a reproduction:

1. Edit the `hecs-plugin-core` package `WorldTransformSystem` and insert a `throw new Error('LOL')` into the `update()` function
1. Run `yarn build` in `hecs-plugin-core`
1. Run `yarn start` in `hecs-example`

This is what I see in the console:

![image](https://user-images.githubusercontent.com/760516/99604860-929b3300-2a5a-11eb-80b0-6e20c93aae98.png)

(we're expecting to see minified code which is useless to us)

I then:

1. Changed `hecs-plugin-core` webpack config to remove `config.devtool = false`
1. Ran `yarn build` in `hecs-plugin-core`
1. Refreshed the browser

This is what I see

![image](https://user-images.githubusercontent.com/760516/99605031-f3c30680-2a5a-11eb-8fd6-679ec6e9a783.png)

All is good!

I also don't think we need to add the `files` change in this PR. By default, Npm deploys all files. If you run `npm pack` you can see what gets published and both `src` and `dist` are included already.

Just for brevity, I also double checked the source-maps are a separate file so as not to increase the bundle size.

**Comment by ashconnell** [comment]
@canadaduane if this works for you, lets remove the `files` change and then I can merge

**Comment by canadaduane** [comment]
Ok, I can confirm this works in `hecs-example`. I'm unable to see sources in my own project, but I suspect this may be my own misconfiguration.
